domains/auth/Auth.API/API/Services/CryptographyService.cs

```diff
@@ -10,39 +10,44 @@ public class CryptographyService : ICryptographyService
 
     public string EncryptState(AuthState state)
     {
-        byte[] iv = new byte[16];
+        byte[] data;
+        //byte[] iv = new byte[16];
         using (Aes aes = Aes.Create())
         {
             aes.Key = Encoding.UTF8.GetBytes(key);
-            aes.IV = iv;
             ICryptoTransform encryptor = aes.CreateEncryptor(aes.Key, aes.IV);
             using (MemoryStream memoryStream = new MemoryStream())
             {
+                memoryStream.Write(aes.IV);
+
                 using (CryptoStream cryptoStream = new CryptoStream(memoryStream, encryptor, CryptoStreamMode.Write))
                 {
                     using (StreamWriter streamWriter = new StreamWriter(cryptoStream))
                     {
                         streamWriter.Write(state.ToString());
                     }
-                    byte[] data = memoryStream.ToArray();
-                    return Convert.ToBase64String(data);
+                    data = memoryStream.ToArray();
+                    
                 }
             }
+            return Convert.ToBase64String(data);
```

Comment:
The reason to wait until after closing the using for cryptostream is to ensure it has flushed its write to the memory streams.

```suggestion
                 }
                 return Convert.ToBase64String(memoryStream.ToArray());
             }
```